File: 1394. Find Lucky Integer in an Array/1394.go
package leetcode

func findLucky(arr []int) int {
	myDict := make(map[int]int, 0)
	
	for _,v := range arr{
		if value, ok := myDict[v]; ok{
			myDict[v] = (value+1)
		}else{
			myDict[v] = 1
		}
	}

	res := -1

	for i,v := range myDict{
		if i==v{
			if v>res{
				res = v
			}
		}
	}

	return res
}

File: 1394. Find Lucky Integer in an Array/_1394.py
class Solution:
    def findLucky(self, arr) -> int:
        import collections
        myDict = collections.Counter(arr)

        for v in myDict.most_common():
            if v[0] == v[1]:
                return v[0]

        return -1




a = Solution()
print(a.findLucky([7,7,7,7,7,7,7]))